src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,208 @@
+/**
+ * @file softshrink.hpp
+ * @author Lakshya Ojha
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+/**
+ * SoftShrinkage operator is defined as,
+ *         | x - lambda, if x >  lambda
+ *  f(x) = | x + lambda, if x < -lambda
+ *         | 0, otherwise
+ *
+ * lambda is set to 0.5 by default.
+ *
+ *          | 1 , if x >  lambda
+ *  f'(x) = | 1 , if x < -lambda
+ *          | 0 , otherwise
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class SoftShrink
+{
+ public:
+  /**
+   * Create SOFTSHRINK object using specified hyperparameter lambda.
+   * Default (lambda = 0.5). lambda should be > 0.
+   * 
+   * @param lambda parameter
+   */
+  SoftShrink(const double lambda = 0.5);
+  /**
```

Comment:
Pendatic style issue, this is missing an empty line.

Reply:
Corrected the same.
Thanks.